In [ ]:
from pandas import *  # pour lire, importer et manipuler des données sous forme tableur
from numpy import *   # pour faire des calculs
from matplotlib.pylab import * # pour faire des graphiques

In [81]:
df = read_csv('vgsales.csv')

df.shape            # nombre de lignes/colonnes
df.info()           # types de chaque colonne, valeurs manquantes
df.describe()       # stats de base sur les colonnes numériques
df.head()           # un aperçu des premières lignes
df.isna().sum()     # combien de valeurs manquantes par colonne

<class 'pandas.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  str    
 2   Platform      16598 non-null  str    
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  str    
 5   Publisher     16540 non-null  str    
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), str(4)
memory usage: 1.4 MB


Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

On remarque que dans ce dataset, les colonnes Year et Publisher ont quelques valeurs manquantes. 

In [82]:
# on va maintennat filtrer les données pour voir les endroits ou il manque des valeurs dans year

year = df["Year"]
publisher = df["Publisher"]

yearVoid = year[year.isna() | (year == "")]
publisherVoid = publisher[publisher.isna() | (publisher == "")]
print(yearVoid)
print(publisherVoid)

print(len(yearVoid))
print(len(publisherVoid))

179     NaN
377     NaN
431     NaN
470     NaN
607     NaN
         ..
16307   NaN
16327   NaN
16366   NaN
16427   NaN
16493   NaN
Name: Year, Length: 271, dtype: float64
470      NaN
1303     NaN
1662     NaN
2222     NaN
3159     NaN
3166     NaN
3766     NaN
4145     NaN
4526     NaN
4635     NaN
5302     NaN
5647     NaN
6272     NaN
6437     NaN
6562     NaN
6648     NaN
6849     NaN
7208     NaN
7351     NaN
7470     NaN
7953     NaN
8330     NaN
8341     NaN
8368     NaN
8503     NaN
8770     NaN
8848     NaN
8896     NaN
9517     NaN
9749     NaN
10382    NaN
10494    NaN
11076    NaN
11526    NaN
12487    NaN
12517    NaN
13278    NaN
13672    NaN
13962    NaN
14087    NaN
14296    NaN
14311    NaN
14698    NaN
14942    NaN
15056    NaN
15261    NaN
15325    NaN
15353    NaN
15788    NaN
15915    NaN
16191    NaN
16198    NaN
16208    NaN
16229    NaN
16367    NaN
16494    NaN
16543    NaN
16553    NaN
Name: Publisher, dtype: str
271
58


On a donc 271 cellules vides dans la colonne Year et 58 colonnes vides dans la colonne publisher. Pour la colonne publisher chaque fois qu'on a un vide on peut mettre "Unknown" pour dire que le publicateur du jeu est inconnu. Pour les années lorsqu'il y a un vide on peut le remplacer par 'N/A' (Non Atribué pour éviter toutes erreurs )

In [92]:
df["Year"] = df["Year"].fillna("N/A")
yearVoid = year[year == "N/A"]
print(yearVoid.head())
print(df["Year"].unique())

179    N/A
377    N/A
431    N/A
470    N/A
607    N/A
Name: Year, dtype: object
[2006.0 1985.0 2008.0 2009.0 1996.0 1989.0 1984.0 2005.0 1999.0 2007.0
 2010.0 2013.0 2004.0 1990.0 1988.0 2002.0 2001.0 2011.0 1998.0 2015.0
 2012.0 2014.0 1992.0 1997.0 1993.0 1994.0 1982.0 2003.0 1986.0 2000.0
 'N/A' 1995.0 2016.0 1991.0 1981.0 1987.0 1980.0 1983.0 2020.0 2017.0]


On a donc remplacé les vides par des "N/A" (Non Attribué) faisons de meme pour la colonne publish

In [93]:
df["Publisher"] = df["Publisher"].fillna("N/A")
publisherVoid = publisher[publisher == "N/A"]
print(publisherVoid.head())
print(df["Publisher"].unique())

470     N/A
1303    N/A
1662    N/A
2222    N/A
3159    N/A
Name: Publisher, dtype: str
<StringArray>
[                   'Nintendo',      'Microsoft Game Studios',
        'Take-Two Interactive', 'Sony Computer Entertainment',
                  'Activision',                     'Ubisoft',
          'Bethesda Softworks',             'Electronic Arts',
                        'Sega',                  'SquareSoft',
 ...
                   'Nitroplus',         'Paradox Development',
                     'Otomate',                   'Ongakukan',
                    'Commseed',                'Inti Creates',
                      'Takuyo',          'Interchannel-Holon',
                  'Rain Games',           'UIG Entertainment']
Length: 579, dtype: str


il y a donc plus de cellule vide dans Publisher et dans Year. On va aussi remplacer le type de la clonne year qui est un float en int